hermes-client/src/main/java/pl/allegro/tech/hermes/client/HermesClient.java

```diff
@@ -1,42 +1,67 @@
 package pl.allegro.tech.hermes.client;
 
+import net.jodah.failsafe.Failsafe;
+import net.jodah.failsafe.RetryPolicy;
+
 import java.net.URI;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.concurrent.CompletableFuture;
-import java.util.concurrent.CompletionStage;
 import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
 import java.util.concurrent.atomic.AtomicInteger;
-import java.util.function.Function;
 import java.util.function.Predicate;
 
-import static java.util.concurrent.CompletableFuture.completedFuture;
-import static java.util.stream.IntStream.range;
 import static pl.allegro.tech.hermes.client.HermesMessage.hermesMessage;
 
 public class HermesClient {
 
     private final HermesSender sender;
     private final String uri;
     private final Map<String, String> defaultHeaders;
-    private final int retries;
-    private final Predicate<HermesResponse> retryCondition;
     private final AtomicInteger currentlySending = new AtomicInteger(0);
+    private final RetryPolicy retryPolicy;
+    private final ScheduledExecutorService scheduler;
     private volatile boolean shutdown = false;
 
     HermesClient(HermesSender sender,
                  URI uri,
                  Map<String, String> defaultHeaders,
                  int retries,
-                 Predicate<HermesResponse> retryCondition) {
+                 Predicate<HermesResponse> retryCondition,
+                 long retrySleepInMillis,
+                 long maxRetrySleepInMillis) {
+        this(sender, uri, defaultHeaders, retries, retryCondition, retrySleepInMillis,
```

Comment:
(i deleted the other comment about executorservice because i noticed the other constructor)
why do we need two constructors? they are package private, nobody should use them except from us